pylint/checkers/classes.py

```diff
@@ -995,16 +995,22 @@ class SpecialMethodsChecker(BaseChecker):
                   'invalid number of parameters. If it has too few or '
                   'too many, it might not work at all.',
                   {'old_names': [('E0235', 'bad-context-manager')]}),
+        'E0303': ('__len__ does not return non-negative integer',
+                  'bad-len-returned',
```

Comment:
I don't like saying "negative int" when returned value may be None or not integer at all. Also I've notices issues like #583, #584, #585, #560 - let's say they  should / shouldn't they share error code, like some generic "invalid-type-returned"?